[CMP-2336]Compliance Operator 1.4.1 Bugs (#74715)
openshift/openshift-docs: security/compliance_operator/compliance-operator-release-notes.adoc

To access the latest release, see xref:../../security/compliance_operator/co-management/compliance-operator-updating.adoc#olm-preparing-upgrade_compliance-operator-updating[Updating the Compliance Operator].

[id="compliance-operator-release-notes-1-4-1"]
== OpenShift Compliance Operator 1.4.1

The following advisory is available for the OpenShift Compliance Operator 1.4.1:

> 

Reviewer: The link for the advisory is not presumably because the release has not gone live yet. Just noting as a reminder to go back and check the link after the release goes live.

Author: You are correct! (I should have noted that in comment 0, which is customary)

* link:https://access.redhat.com/errata/RHBA-2024:1830[RHBA-2024:1830 - OpenShift Compliance Operator bug fix and enhancement update]

[id="compliance-operator-1-4-1-new-features-and-enhancements"]
=== New features and enhancements

* With this update, Compliance Operator now provides `OCP4 STIG ID` and `SRG` with the profile rules. (link:https://issues.redhat.com/browse/CMP-2401[*CMP-2401*])

* With this update, obsolete rules being applied to `s390x` have been removed. (link:https://issues.redhat.com/browse/CMP-2471[*CMP-2471*])

[id="compliance-operator-1-4-1-bug-fixes"]
=== Bug fixes

> * Previously, for RHCOS systems using RHEL9, application of the `upstream-ocp4-kubelet-enable-protect-kernel-sysctl-file-exist` rule would fail. With this update, after auto remediation is applied, RHEL9-based RHCOS systems will show `PASS` upon the application of this rule. (link:https://issues.redhat.com/browse/OCPBUGS-13589[*OCPBUGS-13589*])

Reviewer 2: [According to the SSG](https://redhat-documentation.github.io/supplementary-style-guide/#rhel), we shouldn't use RHEL, unless there's an exception specific to a team's usage. Is there an exception made for this feature?

Author: @bburt-rh good catch, and I worried about that. RHCOS-based containers come in "RHEL9" and "RHEL8" flavor only, and must be chosen according to that designation. We don't have a variable name for that, so I left it.

Reviewer 3: The rule should be `ocp4-kubelet-enable-protect-kernel-sysctl-file-exist` for downstream build.
And the following description is not accurate: "With this update, after auto remediation is applied, RHEL9-based RHCOS systems will show `PASS` upon the application of this rule". We removed this rule, and use  `ocp4-kubelet-enable-protect-kernel-sysctl` instead.

Author: @xiaojiey I have made the requested change, thank you very much.

> * Previously, for RHCOS systems using RHEL9, application of the `ocp4-kubelet-enable-protect-kernel-sysctl-file-exist` rule would fail. This update replaces the rule with `ocp4-kubelet-enable-protect-kernel-sysctl`. Now, after auto remediation is applied, RHEL9-based RHCOS systems will show `PASS` upon the application of this rule. (link:https://issues.redhat.com/browse/OCPBUGS-13589[*OCPBUGS-13589*])

Reviewer 4: ```suggestion
* Previously, for RHCOS systems using RHEL9, application of the `ocp4-kubelet-enable-protect-kernel-sysctl-file-exist` rule failed. This update replaces the rule with `ocp4-kubelet-enable-protect-kernel-sysctl`. Now, after auto remediation is applied, RHEL9-based RHCOS systems show `PASS` upon the application of this rule. (link:https://issues.redhat.com/browse/OCPBUGS-13589[*OCPBUGS-13589*])
```
No future tense (and I would remove would for the sake of minimalism).

> * Previously, after applying compliance remediations using profile `rhcos4-e8`, the nodes are no longer accessible via SSH using the core user account. With this update, nodes remain accessible via SSH using sshkey. (link:https://issues.redhat.com/browse/OCPBUGS-1833[*OCPBUGS-1833*])

Reviewer: You did a very good job of using the same structure in the Bug Fixes section, and I have a small suggestion for your to consider. I think the line describing the previous behavior might read just a little more clearly if it makes use of past tense. For example instead of:

"Previously, after applying compliance remediations using profile `rhcos4-e8`, the nodes are no longer accessible via SSH using the core user account. With this update, nodes remain accessible via SSH using sshkey."

How about something like this:

Previously, after applying compliance remediations using profile `rhcos4-e8`, the nodes were no longer accessible via SSH using the core user account. With this update, nodes remain accessible via SSH using sshkey.

Author: Agreed

> * Previously, after applying compliance remediations using profile `rhcos4-e8`, the nodes were no longer accessible via SSH using the core user account. With this update, nodes remain accessible via SSH using sshkey. (link:https://issues.redhat.com/browse/OCPBUGS-1833[*OCPBUGS-1833*])

Reviewer 4: ```suggestion
* Previously, after applying compliance remediations using profile `rhcos4-e8`, the nodes were no longer accessible through SSH using the core user account. With this update, nodes remain accessible through SSH using sshkey. (link:https://issues.redhat.com/browse/OCPBUGS-1833[*OCPBUGS-1833*])
```
Per https://www.ibm.com/docs/en/ibm-style?topic=word-usage#v on "via". "Through" seemed appropriate, but of course, your call for accuracy.

* Previously, the `STIG` profile was missing rules from CaC that fulfill requirements on the published `STIG` for {product-title}. With this update, upon remediation, the cluster satisfies `STIG` requirements that can be remediated using Compliance Operator. (link:https://issues.redhat.com/browse/OCPBUGS-26193[*OCPBUGS-26193*])

> * Previously, creating a `ScanSettingBinding` with two profiles would override the prohibition against multiple product types. With this update, validation is now consistent regardless of the number of profiles listed in `ScanSettingBinding`. (link:https://issues.redhat.com/browse/OCPBUGS-26229[*OCPBUGS-26229*])

Reviewer 5: I think this note is a little off of what was actually done.

```suggestion
* Previously, creating a `ScanSettingBinding` with profiles of different types for multiple products would bypass a restriction against multiple products types in a binding. With this update, the product validation is relaxed and now allows multiple products regardless of the of profile types in the `ScanSettingBinding`. (link:https://issues.redhat.com/browse/OCPBUGS-26229[*OCPBUGS-26229*])
```

Author: @yuumasato may I omit the word "relaxed" and instead simply say "With this update, the product validation now allows multiple products....."

> * Previously, creating a `ScanSettingBinding` with profiles of different types for multiple products would bypass a restriction against multiple products types in a binding. With this update, the product validation now allows multiple products regardless of the of profile types in the `ScanSettingBinding`. (link:https://issues.redhat.com/browse/OCPBUGS-26229[*OCPBUGS-26229*])

Reviewer 4: ```suggestion
* Previously, creating a `ScanSettingBinding` with profiles of different types for multiple products bypassed a restriction against multiple products types in a binding. With this update, the product validation now allows multiple products regardless of the of profile types in the `ScanSettingBinding`. (link:https://issues.redhat.com/browse/OCPBUGS-26229[*OCPBUGS-26229*])
```

> * Previously, running rule `rhcos4-service-debug-shell-disabled` would show as `FAIL` even after auto-remediation is applied. With this update, rule `rhcos4-service-debug-shell-disabled` now shows `PASS` after auto-remediation is applied. (link:https://issues.redhat.com/browse/OCPBUGS-28242[*OCPBUGS-28242*])

Reviewer: A similar comment about tense for this entry.

Instead of "Previously, running rule `rhcos4-service-debug-shell-disabled` would show as `FAIL` even after auto-remediation is applied. With this update, rule `rhcos4-service-debug-shell-disabled` now shows `PASS` after auto-remediation is applied."

How about:

Previously, running rule `rhcos4-service-debug-shell-disabled` would show as `FAIL` even after auto-remediation was applied. With this update, rule `rhcos4-service-debug-shell-disabled` now shows `PASS` after auto-remediation is applied.

> * Previously, running rule `rhcos4-service-debug-shell-disabled` would show as `FAIL` even after auto-remediation was applied. With this update, rule `rhcos4-service-debug-shell-disabled` now shows `PASS` after auto-remediation is applied. (link:https://issues.redhat.com/browse/OCPBUGS-28242[*OCPBUGS-28242*])

Reviewer 4: ```suggestion
* Previously, running the `rhcos4-service-debug-shell-disabled` rule showed as `FAIL` even after auto-remediation was applied. With this update, running the `rhcos4-service-debug-shell-disabled` rule shows `PASS` after auto-remediation is applied. (link:https://issues.redhat.com/browse/OCPBUGS-28242[*OCPBUGS-28242*])
```
There's a rule for this structure but I've lost the reference for the moment.

* Instructions for the use of rule `rhcos4-banner-etc-issue` have been enhanced to provide more helpful detail. (link:https://issues.redhat.com/browse/OCPBUGS-28797[*OCPBUGS-28797*])

> * Previously the `api_server_api_priority_flowschema_catch_all` rule would provide `FAIL` status on OpenShift 4.16 clusters. With this update, the `api_server_api_priority_flowschema_catch_all` rule provides `PASS` status on OpenShift 4.16 clusters. (link:https://issues.redhat.com/browse/OCPBUGS-28918[*OCPBUGS-28918*])

Reviewer 2: ```suggestion
* Previously the `api_server_api_priority_flowschema_catch_all` rule would provide `FAIL` status on {product-title} 4.16 clusters. With this update, the `api_server_api_priority_flowschema_catch_all` rule provides `PASS` status on {product-title} 4.16 clusters. (link:https://issues.redhat.com/browse/OCPBUGS-28918[*OCPBUGS-28918*])
```
Use an attribute for OpenShift

Author: Ooo, yes. Damb me.

> * Previously the `api_server_api_priority_flowschema_catch_all` rule would provide `FAIL` status on {product-title} 4.16 clusters. With this update, the `api_server_api_priority_flowschema_catch_all` rule provides `PASS` status on {product-title} 4.16 clusters. (link:https://issues.redhat.com/browse/OCPBUGS-28918[*OCPBUGS-28918*])

Reviewer 4: ```suggestion
* Previously the `api_server_api_priority_flowschema_catch_all` rule provided `FAIL` status on {product-title} 4.16 clusters. With this update, the `api_server_api_priority_flowschema_catch_all` rule provides `PASS` status on {product-title} 4.16 clusters. (link:https://issues.redhat.com/browse/OCPBUGS-28918[*OCPBUGS-28918*])
```

> * Previously, when a profile is removed from a completed scan shown in `ScanSettingBinding`, Compliance Operator does not remove the old scan. Afterward, when launching a new SSB using the deleted profile, Compliance Operator fails to update the result. With this update of Compliance Operator, the new SSB now shows the new compliance check result. (link:https://issues.redhat.com/browse/OCPBUGS-29272[*OCPBUGS-29272*])

Reviewer: Similar comment. Instead of:

"Previously, when a profile is removed from a completed scan shown in `ScanSettingBinding`, Compliance Operator does not remove the old scan. Afterward, when launching a new SSB using the deleted profile, Compliance Operator fails to update the result. With this update of Compliance Operator, the new SSB now shows the new compliance check result."

Maybe use "Previously, when a profile is removed from a completed scan shown in `ScanSettingBinding`, the Compliance Operator did not remove the old scan. Afterward, when launching a new SSB using the deleted profile, the Compliance Operator failed to update the result. With this update of  the Compliance Operator, the new SSB shows the compliance check result."

Author: Agreed, done!

> * Previously, when a profile is removed from a completed scan shown in `ScanSettingBinding`, Compliance Operator did not remove the old scan. Afterward, when launching a new SSB using the deleted profile, Compliance Operator failed to update the result. With this release of Compliance Operator, the new SSB now shows the new compliance check result. (link:https://issues.redhat.com/browse/OCPBUGS-29272[*OCPBUGS-29272*])

Reviewer 2: Is the SSB acronym defined somewhere earlier? If not, I would change the first instance to `<full spelled-out name> (SSB)`

Author: SSB is short for "ScanSettingBinding". I'll add it in parenthesis after first use.
